Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -55,10 +69,93 @@ extension OfflineMapAreasView {
+            
+            var mapAreas: [OfflinePreplannedMapArea] = []
+            
+            for mapAreaID in mapAreaIDs {
+                if let preplannedMapAreaID = PortalItem.ID(mapAreaID),
+                   let mapArea = await makeMapArea(
+                    for: portalItemID,
+                    preplannedMapAreaID: preplannedMapAreaID
+                   ) {
+                    mapAreas.append(mapArea)
+                }
+            }
+            
+            offlinePreplannedMapModels = mapAreas.map { mapArea in
+                PreplannedMapModel(
+                    offlineMapTask: offlineMapTask,
+                    mapArea: mapArea,
+                    portalItemID: portalItemID,
+                    preplannedMapAreaID: mapArea.id!
+                )
+            }
+            .sorted(using: KeyPathComparator(\.preplannedMapArea.title))
+            .filter(\.status.isDownloaded)
+        }
+        
+        /// Creates a preplanned map area using a given portal item and map area ID to search for a corresponding
+        /// downloaded mobile map package. If the mobile map package is not found then `nil` is returned.
+        /// - Parameters:
+        ///   - portalItemID: The portal item ID.
+        ///   - preplannedMapAreaID: The preplanned map area ID.
+        /// - Returns: The preplanned map area.
+        private func makeMapArea(
+            for portalItemID: PortalItem.ID,
+            preplannedMapAreaID: PortalItem.ID
+        ) async -> OfflinePreplannedMapArea? {
+            let fileURL = FileManager.default.preplannedDirectory(
+                forPortalItemID: portalItemID,
+                preplannedMapAreaID: preplannedMapAreaID
+            )
+            guard FileManager.default.fileExists(atPath: fileURL.path()) else { return nil }
+            // Make sure the directory is not empty because the directory will exist as soon as the
+            // job starts, so if the job fails, it will look like the mmpk was downloaded.
+            guard !FileManager.default.isDirectoryEmpty(atPath: fileURL) else { return nil }
+            let mmpk = MobileMapPackage(fileURL: fileURL)
+            
+            try? await mmpk.load()
+            guard let item = mmpk.item else { return nil }
+            
+            return OfflinePreplannedMapArea(
+                title: item.title,
+                description: item.description,
+                id: preplannedMapAreaID,
+                thumbnail: item.thumbnail
+            )
         }
     }
 }
+
+private struct OfflinePreplannedMapArea: PreplannedMapAreaProtocol {
+    var packagingStatus: PreplannedMapArea.PackagingStatus?
+    var title: String
+    var description: String
+    var thumbnail: LoadableImage?
+    var id: PortalItem.ID?
+    
+    func retryLoad() async throws {}
+    func makeParameters(using offlineMapTask: OfflineMapTask) async throws -> DownloadPreplannedOfflineMapParameters {
+        throw CancellationError()
+    }
```

Comment:
I don't have a suggestion, but I'd like to know other reviewer's idea on this.

Reply:
Oh I mean we have a protocol method to implement for the local map areas, but it is highly likely that this would never be called. Should it be a `fatalError` or similar kind?

Reply:
If you're sure that the method won't be called, then fatal error is fine.